waku/v2/node/keepalive.go

```diff
@@ -81,36 +90,45 @@ func (w *WakuNode) startKeepAlive(ctx context.Context, randomPeersPingDuration t
 				continue
 			}
 
-			// Priorize mesh peers
-			meshPeersSet := make(map[peer.ID]struct{})
-			for _, t := range w.Relay().Topics() {
-				for _, p := range w.Relay().PubSub().MeshPeers(t) {
-					meshPeersSet[p] = struct{}{}
-				}
-			}
-			peersToPing = append(peersToPing, maps.Keys(meshPeersSet)...)
-
-			// Ping also some random relay peers
-			if maxPeersToPing-len(peersToPing) > 0 {
-				relayPeersSet := make(map[peer.ID]struct{})
+			if w.opts.enableRelay {
+				// Priorize mesh peers
+				meshPeersSet := make(map[peer.ID]struct{})
 				for _, t := range w.Relay().Topics() {
-					for _, p := range w.Relay().PubSub().ListPeers(t) {
-						if _, ok := meshPeersSet[p]; !ok {
-							relayPeersSet[p] = struct{}{}
-						}
+					for _, p := range w.Relay().PubSub().MeshPeers(t) {
+						meshPeersSet[p] = struct{}{}
 					}
 				}
+				peersToPing = append(peersToPing, maps.Keys(meshPeersSet)...)
+
+				// Ping also some random relay peers
+				if maxPeersToPingPerProtocol-len(peersToPing) > 0 {
+					relayPeersSet := make(map[peer.ID]struct{})
+					for _, t := range w.Relay().Topics() {
+						for _, p := range w.Relay().PubSub().ListPeers(t) {
+							if _, ok := meshPeersSet[p]; !ok {
+								relayPeersSet[p] = struct{}{}
+							}
+						}
+					}
 
-				relayPeers := maps.Keys(relayPeersSet)
-				rand.Shuffle(len(relayPeers), func(i, j int) { relayPeers[i], relayPeers[j] = relayPeers[j], relayPeers[i] })
+					relayPeers := maps.Keys(relayPeersSet)
+					rand.Shuffle(len(relayPeers), func(i, j int) { relayPeers[i], relayPeers[j] = relayPeers[j], relayPeers[i] })
 
-				peerLen := maxPeersToPing - len(peersToPing)
-				if peerLen > len(relayPeers) {
-					peerLen = len(relayPeers)
+					peerLen := maxPeersToPingPerProtocol - len(peersToPing)
+					if peerLen > len(relayPeers) {
+						peerLen = len(relayPeers)
+					}
+					peersToPing = append(peersToPing, relayPeers[0:peerLen]...)
 				}
-				peersToPing = append(peersToPing, relayPeers[0:peerLen]...)
 			}
 
+			// We also ping all filter nodes
+			filterPeersSet := make(map[peer.ID]struct{})
```

Comment:
shouldn't this be in else condition of `w.opts.enableRelay`?

Reply:
go-waku does allow a node to be a filter client and also use relay at the same time (even if it does not make sense :thinking: )

Reply:
Ah right, i remember at some point discussing about this as a potential possibility.